spec/controllers/users_controller_spec.rb

```diff
@@ -18,4 +18,18 @@
       expect(count).to eq 0
     end
   end
+
+  describe 'GET activate' do
+    let(:user) { create(:user, password: 'valid') }
+
+    it 'logs in user from correct token' do
+      get :activate, id: user.activation_token
+      expect(controller.current_user).to eq user.reload
+    end
+
+    it 'fails to activate user from invalid token' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.